packages/web-react/src/components/FileUploader/useFileUploaderInput.ts

```diff
@@ -0,0 +1,162 @@
+    if (file.size > maxFileSize) {
+      throw new Error(errorMessages?.errorMaxFileSize);
+    }
+  };
+
+  const checkFileDuplicity = (file: File) => {
+    if (isMultiple && fileQueue.has(getUpdatedFileName(file.name))) {
+      throw new Error(errorMessages?.errorFileDuplicity);
+    }
+  };
+
+  const checkQueueSize = () => {
+    if (fileQueue.size >= maxUploadedFiles) {
+      throw new Error(errorMessages?.errorMaxUploadedFiles);
+    }
+  };
+
+  const checkAllowedFileType = (file: File) => {
+    const fileExtension = file.name.split('.').pop()?.toLowerCase() as string;
+    const fileType = file.type;
+    let isSupported;
+    // Determine if validation should occur
+    // If the `accept` attribute is set and not all types are allowed
+    const shouldValidate = accept && !(accept === '' || accept === '*' || accept === '*/*');
+
+    if (!shouldValidate) {
+      isSupported = true;
+    } else {
+      // An array of all allowed file types and extensions
+      const acceptArray = accept.replace(/ /g, '').split(',');
+      // An array of all allowed file extensions
+      const acceptExtensions = acceptArray.filter((exp) => exp.match(/\./g));
+      // An array of all allowed file types
+      const acceptTypes = acceptArray.filter((exp) => exp.match(/\//g));
+
+      acceptExtensions.forEach((exp) => {
+        // Modified expression for better comparison
+        const expression = exp.replace('.', '').replace('*', '');
+
+        if (fileExtension.match(expression)) {
+          isSupported = true;
+        }
+      });
+
+      acceptTypes.forEach((exp) => {
+        // Modified expression for better comparison
+        const expression = exp.replace('*', '');
+
+        if (fileType.match(expression)) {
+          isSupported = true;
+        }
+      });
+    }
+
+    if (!isSupported) {
+      throw new Error(`"${file.name}" is not supported file`);
+    }
+  };
+
+  const clearQueueHandler = () => {
+    if (!isMultiple) {
+      clearQueue();
+    }
+  };
+
+  const fileProcessHandler = (file: File) => {
+    try {
+      clearQueueHandler();
+      checkFileDuplicity(file);
+      checkAllowedFileSize(file);
+      checkAllowedFileType(file);
+      checkQueueSize();
+      addToQueue(getUpdatedFileName(file.name), file);
+    } catch (error) {
+      if (onError) {
+        onError(error);
+      } else {
+        // We want to show error message in console
+        /* eslint-disable-next-line no-console */
+        console.warn(error);
+      }
+    }
+  };
+
+  const onDropHandler = (event: DragEvent<HTMLDivElement>) => {
+    const transferItems = Array.from(event.dataTransfer.items);
+    const transferFiles = Array.from(event.dataTransfer.files);
+
+    let counter = 0;
+    counter += fileQueue.size;
+
+    if (event.dataTransfer.items) {
+      transferItems.forEach((item) => {
+        if (item.kind === 'file') {
+          const file = item.getAsFile();
+          if (file) {
```

Comment:
Can we merge these two conditions into one, please? `if(file && counter < maxUploadedFiles)`.

Reply:
Done